src/main/java/org/sonar/plugins/checkstyle/metadata/CheckstyleMetadata.java

```diff
@@ -314,10 +314,18 @@ private static String getRuleTag(String checkPackage,
     public static String getFullCheckName(String checkName) {
         final int capacity = 1024;
         final StringBuilder result = new StringBuilder(capacity);
-        for (int i = 0; i < checkName.length(); i++) {
-            result.append(checkName.charAt(i));
-            if (i + 1 < checkName.length() && Character.isUpperCase(checkName.charAt(i + 1))) {
-                result.append(' ');
+        if ("JavaNCSS".equals(checkName)) {
```

Comment:
can we move this to a map:
JavaNCSS -> Java NCSS
NPathComplexity -> NPath Complexity

if we find the key in the map, append the value, otherwise get the name using the for loop.

this allows us to very easily add other such cases with kinda 1 line of code. :)

Reply:
@muhlba91 Done.